packages/realm-web/src/utils/ejson.ts

```diff
@@ -34,9 +38,9 @@ export function serialize(obj: object): object {
  * @param obj The object or array of objects in extended-JSON format.
  * @returns The object or array of objects with inflated BSON types.
  */
-export function deserialize<R extends any>(obj: object | object[]): R {
+export function deserialize<R extends any>(obj: object | object[]): R | R[] {
     if (Array.isArray(obj)) {
-        return obj.map(doc => EJSON.deserialize(doc)) as any;
+        return obj.map(doc => EJSON.deserialize(doc)) as R[];
```

Comment:
I don't like this change. With this it's no longer possible to lock down the return type to not return an array.
It was the intent that if an array was returned, `R` needed to be an array and the consumer of the API was responsible for providing the correct return type (which was potentially an array of something).